### PR TITLE
fix memory speed

### PR DESCRIPTION
**Pull request by Sakoram**

### CODE REVIEW

File: src/mainboard/pcengines/apu2/mainboard.c [hunk 1]
```diff
@@ -237,6 +237,24 @@ static int mainboard_smbios_type16(DMI_INFO *agesa_dmi, int *handle,
...
+		case 166:  return 333;
+		case 216:  return 433;
+		case 266:  return 533;
+		case 333:  return 667;
+		case 556:  return 1111;
+		case 667:  return 1333;
+		case 688:  return 1375;
+		case 813:  return 1625;
+		case 1066: return 2133;
+		case 1666: return 3333;
```
Comment (miczyg1) [inline src/mainboard/pcengines/apu2/mainboard.c:253]: @Sakoram don't you see any pattern? It is either `bus*2` or `bus*2 + 1` so you may use following notation:
```
                case 166: 
		case 216: 
		case 266:
		case 333:
		case 556:
		case 667:
		case 688:
		case 813:
		case 1066:
		case 1666:return bus * 2 + 1;
```

File: src/mainboard/pcengines/apu2/mainboard.c [hunk 2]
```diff
@@ -262,15 +280,16 @@ static int mainboard_smbios_type17(DMI_INFO *agesa_dmi, int *handle,
...
+	t->speed = ddr_speed_from_bus_speed(agesa_dmi->T17[0][0][0].Speed);
 	t->manufacturer = agesa_dmi->T17[0][0][0].ManufacturerIdCode;
 	t->serial_number = smbios_add_string(t->eos,
 				agesa_dmi->T17[0][0][0].SerialNumber);
 	t->part_number = smbios_add_string(t->eos,
 				agesa_dmi->T17[0][0][0].PartNumber);
 	t->attributes = agesa_dmi->T17[0][0][0].Attributes;
 	t->extended_size = agesa_dmi->T17[0][0][0].ExtSize;
-	t->clock_speed = agesa_dmi->T17[0][0][0].ConfigSpeed;
+	t->clock_speed = ddr_speed_from_bus_speed(
```
Comment (miczyg1) [inline src/mainboard/pcengines/apu2/mainboard.c:291]: See this field is a clock speed and it should remain unchanged? What is the value of agesa_dmi->T17[0][0][0].ConfigSpeed? Section 7.18:
https://www.dmtf.org/sites/default/files/standards/documents/DSP0134_3.1.1.pdf

Comment (Sakoram) [reply src/mainboard/pcengines/apu2/mainboard.c:291]: Both of these values were 2x bigger before version v4.12.0.1. (You can see it in files attached to issue https://github.com/pcengines/apu2-documentation/issues/176 )
and as far as I understand datasheet both of these values are speed of the memory and are given in MT/s. Only the speed of the bus which would be in MHz should remain unchanged.

File: src/mainboard/pcengines/apu2/mainboard.c (hunk 1 above)
Comment (Sakoram) [reply src/mainboard/pcengines/apu2/mainboard.c]: Actually, some of them were bus*2-1. I've corrected it.

File: src/mainboard/pcengines/apu2/mainboard.c (hunk 2 above)
Comment (miczyg1) [reply src/mainboard/pcengines/apu2/mainboard.c:291]: Yes, becasue the were manually set.  The ConfigSpeed should represent the configured memory clock on the memory bus, so here the conversion is not desired.

**Pull request merged**